sdk/src/main/java/io/dapr/config/Properties.java

```diff
@@ -18,7 +19,7 @@
   /**
    * Dapr's default IP for HTTP and gRPC communication.
    */
-  private static final String DEFAULT_SIDECAR_IP = "127.0.0.1";
+  private static final String DEFAULT_SIDECAR_IP = InetAddress.getLoopbackAddress().getHostAddress();

```

Comment:
This logic might not work. For example, the loopback address might be ::1 but Dapr is listening on 127.0.0.1 only. I think this needs to be a logic to auto-detect where Dapr sidecar is running instead. So, this property would change to something like "" or "auto" and add a logic to auto detect where the sidecar is running.

Reply:
Good point on why loopback address might not work. However, I'm not entirely sure that we need to auto detection logic to detect where the sidecar is running for this, as `DEFAULT_SIDECAR_IP` is only used as a fallback if the `dapr.sidecar.ip` or `DAPR_SIDECAR_IP` System properties are not available which as I understand by definition tells us where the sidecar is running. Thus, I think we just need something reasonable to use for the fallback `DEFAULT_SIDECAR_IP`, so what if we use `InetAddress.getLocalHost()` as the fallback default instead of the hardcoded "127.0.0.1"? Is it reasonable to assume that the Dapr sidecar is running on the same localhost as the application if the sidecar ip properties are not defined?

Reply:
DAPR_SIDECAR_IP env is only used as a workaround for when the default behavior does not work. We want to have the default to be a very good guess to avoid forcing users to set DAPR_SIDECAR_IP for their Java App.

Reply:
I've just looked at the current code in master for runtime and this is the default behavior of the sidecar:
+ Standalone mode: listen to all addresses
+ K8s mode: listen to [::1],127.0.0.1

So, this approach is fine as of now. Let's keep it. User can override this setting if he/she decided to override the default listen ports.

Reply:
@artursouza - awesome, thanks for looking into this. I'd like to confirm here before proceeding:
1. Are you saying that the approach of setting DEFAULT_SIDECAR_IP to the loopback address via `InetAddress.getLoopbackAddress()` is fine?
2. Or are you thinking we need to set DEFAULT_SIDECAR_IP to the localhost address via `InetAddress.getLocalHost()`?

Given the default behavior of the sidecar you've determined, I believe that Option 1 would be the correct approach given that it listens to the loopback localhost addresses in K8s mode. Please **let me know if you also agree with Option 1**, thanks!

Reply:
could have a little bit of logic here depending on the system properties `java.net.preferIPv4Stack` and `java.net.preferIPv6Addresses`

Reply:
@johnou - thanks for the suggestion, though I do wonder if this extra logic is really needed for `DEFAULT_SIDECAR_IP` since the user can override the default IP address for their sidecar through the `dapr.sidecar.ip` property or `DAPR_SIDECAR_IP` env variable, which can be either IPv4 or IPv6 based on their preference. Does this make sense, or am I interpreting your feedback incorrectly?

Reply:
makes sense, I'd keep it simple given they can override the address.